### PR TITLE
Add script to build standalone image with examples

### PR DESCRIPTION
**Pull request by kanterov**

### CODE REVIEW

File: jflyte/src/main/java/org/flyte/jflyte/Execute.java [hunk 1]
```diff
@@ -119,6 +117,38 @@ private void execute() {
     }
   }
 
+  // We support two modes: using staged class path (--indexFileLocation), or using standalone docker
+  // image
+  // and specifying FLYTE_INTERNAL_MAIN_MODULE.
```
Comment (narape) [inline jflyte/src/main/java/org/flyte/jflyte/Execute.java:122]:
```suggestion
  // image and specifying FLYTE_INTERNAL_MAIN_MODULE.
```

File: jflyte/src/main/java/org/flyte/jflyte/Execute.java [hunk 2]
```diff
@@ -119,6 +117,38 @@ private void execute() {
...
+  private ClassLoader loadPackage(
```
Comment (narape) [inline jflyte/src/main/java/org/flyte/jflyte/Execute.java:123]: This method is hard to follow
I think that it would have been more readable to to just right high level code here:

```java
if (indexFileLocation == null && config.mainModule() == null) {
  throw new IllegalArgumentException(
            "FLYTE_INTERNAL_MAIN_MODULE env. variable or --indexFileLocation "
                + "must be set specified. If you are seeing this error, likely something "
                + "is wrong with docker image packaging.");
} else if (indexFileLocation != null) {
  return loadStagedPackage(fileSystems, indexFileLocation);
}
return loadLocalPackage(config, modules);
```

And then write two small methods: `loadStagedPackage` and `loadLocalPackage`

Comment (narape) [reply jflyte/src/main/java/org/flyte/jflyte/Execute.java:123]: Actually I think that it would be cleaner to have this logic in separate classes sharing a common interface, the same for registration

**Comment by kumare3** [comment]
This build is failing for me, maybe I am doing something wrong.
I ran ```/scripts/build_standalone_examples.sh```

**Comment by kumare3** [comment]
> @kumare3 can you share an error message? or run with `bash -x`

```logs
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.921 s
[INFO] Finished at: 2020-08-20T16:19:55-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.2.0:jar (attach-javadocs) on project flytekit-api: MavenReportException: Error while generating Javadoc:
[ERROR] Exit code: 1 - /Users/kumare/go/src/github.com/spotify/flytekit-java/flytekit-api/target/generated-sources/annotations/org/flyte/api/v1/AutoValue_LiteralType.java:49: error: cannot find symbol
[ERROR]   static final class Builder extends LiteralType.Builder {
[ERROR]                                                 ^
[ERROR]   symbol:   class Builder
[ERROR]   location: class LiteralType
[ERROR] /Users/kumare/go/src/github.com/spotify/flytekit-java/flytekit-api/target/generated-sources/annotations/org/flyte/api/v1/AutoValue_LiteralType.java:54: error: cannot find symbol
[ERROR]     public LiteralType.Builder simpleType(SimpleType simpleType) {
[ERROR]                       ^
[ERROR]   symbol:   class Builder
[ERROR]   location: class LiteralType
[ERROR]
[ERROR] Command line was: /usr/local/opt/openjdk/bin/javadoc @options @packages
[ERROR]
[ERROR] Refer to the generated Javadoc files in '/Users/kumare/go/src/github.com/spotify/flytekit-java/flytekit-api/target/apidocs' dir.
```

**Comment by kanterov** [comment]
@kumare3 what java version do you use? Works for me (and CI):
```
$ java -version
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
```

**Comment by kumare3** [comment]
> @kumare3 what java version do you use? Works for me (and CI):
> 
> ```
> $ java -version
> openjdk version "11.0.2" 2019-01-15
> OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
> OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
> ```

This could be a problem, 
```)java -version
java version "10.0.1" 2018-04-17
Java(TM) SE Runtime Environment 18.3 (build 10.0.1+10)
Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10.0.1+10, mixed mode)```

I will try upgrading

**Comment by kanterov** [comment]
I think we do enable TLS based on options, but it doesn't work due to classloader tricks we are doing. It's special because TLS must be using the native library.

**Comment by kumare3** [comment]
@kanterov btw, we are also passing in `--raw-output-prefix` to indicate where the data should be generated, and this is configurable per launch plan. Again users are free to choose their path, but this makes it possible to create sandboxes that are unique per-execution and immutable. Thus if you have network issues (racing etc) or retries then partial data will not be commited as all data is generated in unique paths - https://github.com/lyft/flytekit/commit/1428b3777a2ff998142ee9af7e61cc520fca8fe0

We should discuss if you want to support it in flytekit-java

**Comment by kanterov** [comment]
@kumare3 fixed TLS, can you try again?

**Comment by kumare3** [comment]
> @kumare3 fixed TLS, can you try again?

Wow, will do after the long weekend

**Comment by kumare3** [comment]
@kanterov also Identical task found should not raise an alert ```io.grpc.StatusRuntimeException: ALREADY_EXISTS: identical task already exists with id resource_type:TASK project:"flytesnacks" domain:"development" name:"org.flyte.examples.BatchLookUpTask" version:"8ba0fcf"
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:240)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:221)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:140)
	at flyteidl.service.AdminServiceGrpc$AdminServiceBlockingStub.createTask(AdminServiceGrpc.java:2411)
	at org.flyte.jflyte.FlyteAdminClient.createTask(FlyteAdminClient.java:76)
	at org.flyte.jflyte.RegisterWorkflows.registerAll(RegisterWorkflows.java:218)
	at org.flyte.jflyte.RegisterWorkflows.call(RegisterWorkflows.java:112)
	at org.flyte.jflyte.RegisterWorkflows.call(RegisterWorkflows.java:54)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1783)
	at picocli.CommandLine.access$900(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2150)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2144)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2108)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:1975)
	at picocli.CommandLine.execute(CommandLine.java:1904)
	at org.flyte.jflyte.Main.main(Main.java:68)```

**Comment by kumare3** [comment]
@kanterov 
aah I found the problem, its very small. How do I add https://github.com/lyft/flytesnacks/blob/master/cookbook/notebook.config#L6
https://github.com/lyft/flyteidl/blob/c3baba8983019680ef57b6244cea36ba951233ed/protos/flyteidl/admin/launch_plan.proto#L62

**Comment by honnix** [comment]
Closing stale PR.
